clean-leather.lic

```diff
@@ -2,80 +2,156 @@
+    # them, then putting them in the storage container.
+    loop do
+      case DRC.bput("get #{animal} #{hide} from my #{source}", /^You get/i, /^You carefully remove/i, /^What were you/i, /What do you want to get?/i)
+      when /^You get/i, /^You carefully remove/i
+        scrape_leather(hide, speed)
+        apply_preservative(animal, hide, storage, full_preservative)
+      when /^What were you/i, /What do you want to get?/i
+        DRC.message("Done with all #{animal} #{hide}s.")
+        break
+      else
+        DRC.message("Unable to get the rawhide. Please report this error.")
+        DRCI.stow_hands
       end
-      bput("pour #{@preservative} on my #{args.noun}", 'roundtime')
-      waitrt?
+    end
+  end
 
-      if @bag.nil?
-        fput("stow my #{@preservative}")
+  def scrape_leather(hide, speed)
+    DRCC.get_crafting_item('scraper', @bag, @bag_items, @belt)
+
+    # Scrapes a skin until it's fully-scraped
+    loop do
+      case DRC.bput("scrape #{hide} with my scraper #{speed}", /^You (quickly|carefully)? scrape your/i, /looks as clean as you/i)
+      when /^You (quickly|carefully)? scrape your/i
+        waitrt?
+        next
+      when /looks as clean as you/
+        break
       else
-        fput("put my #{@preservative} in my #{@bag}")
+        DRC.message("Unable to scrape leather. Please report this error.")
+        DRCI.stow_hands
+        exit
+      end
+    end
+
+    DRCC.stow_crafting_item('scraper', @bag, @belt)
+  end
+
+  def apply_preservative(animal, hide, storage, full_preservative)
+
+    # This loop applies a single application of preservative unless
+    # the user has specified "full" application in the args, in which case
+    # it applies until game messages suggest we're at max.
+    loop do
+      # If we have no preservative on-hand, get money,
+      # get preservative, go back to crafting spot.
+      unless DRCI.get_item_if_not_held?(@preservative)
+        DRCM.ensure_copper_on_hand(2000, @settings)
+        DRCT.order_item(@stock_room, @order_number)
+        DRCT.walk_to(@room)
       end
 
-      if args.storage
-        fput("put my #{args.noun} in my #{args.storage}")
+      case DRC.bput("pour #{@preservative} on my #{animal} #{hide}", /^You pour/i, /would only be damaged by adding more/i, /What do you want/i, /^Pour what?/i)
+      when /^You pour/i
+        waitrt?
+        # If user hasn't specified that we apply max preservative, break out
+        # here after a single application.
+        break if !full_preservative
+        next
+      when /^Pour what?/i
+        # We've run out of lotion during the pouring process, re-run method, which will buy more
+        apply_preservative(animal, hide, storage, full_preservative)
+      when /would only be damaged by adding more/i
+        break
       else
-        fput("stow my #{args.noun}")
+        DRC.message("Unable to apply preservative. Please report this error.")
```

Comment:
Can we provide more guidance on _how_ to report this error?

Reply:
Done.